src/Microsoft.IdentityModel.Protocols.WsFederation/WsFederationMetadataSerializer.cs

```diff
@@ -110,6 +126,9 @@ protected virtual void ReadEntityDescriptor(WsFederationConfiguration configurat
         /// <param name="reader">xmlreader</param>
         protected virtual void ReadKeyDescriptorForSigning(WsFederationConfiguration configuration, XmlReader reader)
         {
+            if (configuration == null)
+                throw LogHelper.LogArgumentNullException(nameof(configuration));
+
             XmlUtil.CheckReaderOnEntry(reader, WsFederationConstants.Elements.KeyDescriptor, WsFederationConstants.Namespaces.MetadataNamespace);
 
             if (!IsKeyDescriptorForSigning(reader))
```

Comment:
We should get rid of the method IsKeyDescriptorForSigning. We know we are on a KeyDescriptor element.
If the Attribute "use" is missing, log a warning. If the attribute equals anything but 'signing' throw.